lib/active_model/serializer/adapter/flat_json.rb

```diff
@@ -0,0 +1,17 @@
+module ActiveModel
+  class Serializer
+    module Adapter
+      class FlatJson < JsonApi
+        def serializable_hash(options = {})
```

Comment:
suggestion from @beauby in order to maintain parity with the json api spec regarding deserialization

```ruby
primary_types = document['data'].map {|x| x['type']}
included_types = document['included'].map {|x| x['type']}
types = (primary_types + included_types).uniq
​
hash = {}
types.each do |t|
	primary_collection = deserialize(document['data'].filter { |x| x['type'] == t })
	included_collection = #something similar
	hash[t] = primary_collection + included_collection if unless primary_collection.empty && included_collection.any?
end
```